modules/articles/client/config/articles.client.routes.js

```diff
@@ -12,22 +12,57 @@ angular.module('articles').config(['$stateProvider',
       })
       .state('articles.list', {
         url: '',
-        templateUrl: 'modules/articles/client/views/list-articles.client.view.html'
+        templateUrl: 'modules/articles/client/views/list-articles.client.view.html',
+        controller: 'ArticlesListController',
+        resolve:{
+          articles: function (Articles, $stateParams, $state) {
```

Comment:
In my opinion, `resolve`'s on a route are better suited for when you're interacting with an instance of a resource/entity. When a view is dependent on specific data actually existing at the time of loading. In the case of lists, it's not critical that there's data when the view loads or even if such data exists.

The concern of the route config (`resolve` in this case) should be to make sure the view has everything it needs in order to function properly, and for authorization purposes.